caravel/assets/javascripts/SqlLab/components/SqlEditor.jsx

```diff
@@ -56,36 +56,63 @@ class SqlEditor extends React.Component {
   }
   startQuery(runAsync = false, ctas = false) {
     const that = this;
-    const query = {
-      id: shortid.generate(),
+    let query = {
       sqlEditorId: this.props.queryEditor.id,
       sql: this.props.queryEditor.sql,
-      state: 'running',
       tab: this.props.queryEditor.title,
       dbId: this.props.queryEditor.dbId,
-      startDttm: new Date(),
     };
-    const url = '/caravel/sql_json/';
-    const data = {
+    const createQueryRequest = {
       sql: this.props.queryEditor.sql,
       database_id: this.props.queryEditor.dbId,
+      sql_editor_id: this.props.queryEditor.id,
       schema: this.props.queryEditor.schema,
+      tab: this.props.queryEditor.title,
       json: true,
-      async: runAsync,
       select_as_cta: ctas,
     };
-    this.props.actions.startQuery(query);
+    const createQueryUrl = '/caravel/create_query/';
     $.ajax({
       type: 'POST',
       dataType: 'json',
-      url,
-      data,
+      url: createQueryUrl,
+      data: createQueryRequest,
       success(results) {
-        if (runAsync) {
-          // TODO nothing?
-        } else {
-          that.props.actions.querySuccess(query, results);
-        }
+        query = Object.assign({}, query, results['query'])
```

Comment:
I'll init query right here from the server - const query = Object.assign({}, results['query'])
removed other assignments as they are not needed.
I am not sure how to let the reducer manage the objectids.
Could you please explain me in a bit more details ?